Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py

```diff
@@ -0,0 +1,823 @@
+    """
+    Funtion to decide which normalization to use
+    :param obj: object that we want to normalize
+    :return:
+    """
+    if isinstance(obj, str):
+        return normalize_command_line(obj)
+    else:
+        return normalize_json(obj)  # type: ignore
+
+
+def normalize_json(obj) -> str:  # type: ignore
+    """
+    Normalize json from removing unwantd regex pattern or stop word
+    :param obj:Dumps of a json or dict
+    :return:
+    """
+    if isinstance(obj, float) or not obj:
+        return " "
+    if isinstance(obj, str):
+        obj = json.loads(obj)
+    if check_list_of_dict(obj):
+        obj = {k: v for k, v in enumerate(obj)}  # type: ignore
+    if not isinstance(obj, dict):
+        return " "
+    my_dict = recursive_filter(obj, REGEX_DATE_PATTERN, "None", "N/A", None, "")
+    my_string = json.dumps(my_dict)
+    pattern = re.compile('([^\s\w]|_)+')
+    my_string = pattern.sub(" ", my_string)
+    my_string = my_string.lower()
+    return my_string
+
+
+def normalize_command_line(command) -> str:
+    """
+    Normalize command line
+    :param command: command line
+    :return: Normalized command line
+    """
+    if command and isinstance(command, list):
+        command = ' '.join(set(command))
+    if command and isinstance(command, str):
+        my_string = command.lower()
+        my_string = "".join([REPLACE_COMMAND_LINE.get(c, c) for c in my_string])
+        my_string = my_string.strip()
+        return my_string
+    else:
+        return ''
+
+
+class Tfidf(BaseEstimator, TransformerMixin):
+    """
+    TFIDF transformer
+    """
+
+    def __init__(self, normalize_function):
+        """
+        :param model_params: parameters of TFIDF
+        :param normalize_function: Normalize function to apply on each sample of the corpus before the vectorization
+        """
+        self.normalize_function = normalize_function
+        self.vec = TfidfVectorizer(**TFIDF_PARAMS)
+
+    def fit(self, x, y=None):
+        """
+        Fit TFIDF transformer
+        :param x: incident on which we want to fit the transfomer
+        :return: self
+        """
+        feature_name = x.columns[0]
+        if self.normalize_function:
+            x = x[feature_name].apply(self.normalize_function)
+        self.vec.fit(x)
+        return self
+
+    def transform(self, x):
+        """
+        Transform x with the trained vectorizer
+        :param x: DataFrame or np.array
+        :return:
+        """
+        feature_name = x.columns[0]
+        if self.normalize_function:
+            x = x[feature_name].apply(self.normalize_function)
+        else:
+            x = x[feature_name]
+        return self.vec.transform(x).toarray()
+
+
+def store_model_in_demisto(model: Type[PostProcessing], model_name: str, model_override: bool) -> None:
+    model_data = base64.b64encode(pickle.dumps(model)).decode('utf-8')
+    res = demisto.executeCommand('createMLModel', {'modelData': model_data,
```

Comment:
@yguezpa @erezh31 
Customer who will use this script and the ML page will have an identified record of a failed model
@erezh31 - any ideas if we can hide it? or at least change its "training failed" annoying status?
@yguezpa - so customers will understand what is this record, perhaps add prefix to all model names such as "prefix_" and add a default name in case it's left empty 

![image](https://user-images.githubusercontent.com/57318201/117263721-39ce2400-ae5b-11eb-8d11-a1fd429c76b6.png)

Reply:
@lperryp yes I added option to hide it, @yguezpa please add argument "modelHidden" and set to "true". @lperryp it will work only on latest master

Reply:
@erezh31 thanks
What do you both think of adding "clustering_" prefix to model name? So customers won't be confused until they upgrade?